setup.go

```diff
@@ -58,38 +99,44 @@ func SetupLogging() {
 		zapCfg.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
 	case "json":
 		zapCfg.Encoding = "json"
-	default:
-		zapCfg.Encoding = "console"
-		zapCfg.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
 	}
 
-	zapCfg.Sampling = nil
-	zapCfg.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
-
-	zapCfg.OutputPaths = []string{"stderr"}
 	// check if we log to a file
 	if logfp := os.Getenv(envLoggingFile); len(logfp) > 0 {
 		zapCfg.OutputPaths = append(zapCfg.OutputPaths, logfp)
 	}
 
-	// set the backend(s)
-	lvl := LevelError
-
 	logenv := os.Getenv(envLogging)
 	if logenv == "" {
 		logenv = os.Getenv(envIPFSLogging)
 	}
 
+	// set the backend(s)
+	var lvl LogLevel
 	if logenv != "" {
 		var err error
 		lvl, err = LevelFromString(logenv)
 		if err != nil {
 			fmt.Println("error setting log levels", err)
 		}
+		zapCfg.Level.SetLevel(zapcore.Level(lvl))
+		SetAllLoggers(lvl)
+	}
+}
+
+// SetFieldOnAllLoggers adds the provided key value args as fields to
+// all loggers.
+// Must be called before Logger, i.e. in an init() function.
```

Comment:
We usually construct loggers in the global scope. I'm not sure if we can use this safely.